client/app/pages/dashboards/ShareDashboardDialog.jsx

```diff
@@ -35,7 +35,7 @@ class ShareDashboardDialog extends React.Component {
     };
 
     this.apiUrl = replace(API_SHARE_URL, '{id}', dashboard.id);
-    this.disabled = this.props.hasQueryParams && !dashboard.publicAccessEnabled;
+    this.disabled = !(this.props.hasOnlySafeQueries || dashboard.publicAccessEnabled);
```

Comment:
If I understand this correctly, "Allow public access" switch should be enabled if:

- it is already On;
- it is Off and there are only safe parameters.

My opinion is that it's better to avoid using negated condition (rewrite it as `this.isEnabled = dashboard.publicAccessEnabled || this.props.hasOnlySafeQueries;`) and negate it where it's used - it will be much easier to understand it.

Reply:
Yeah, makes much more sense 👍